go-controller/pkg/ovn/ovn.go

```diff
@@ -112,20 +134,186 @@ func NewOvnController(kubeClient kubernetes.Interface, wf *factory.WatchFactory)
 		gatewayCache:             make(map[string]string),
 		loadbalancerClusterCache: make(map[string]string),
 		loadbalancerGWCache:      make(map[string]string),
+		serviceVIPToName:         make(map[ServiceVIPKey]types.NamespacedName),
+		serviceVIPToNameLock:     sync.Mutex{},
 	}
 }
 
-// Run starts the actual watching.
-func (oc *Controller) Run() error {
-	for _, f := range []func() error{oc.WatchNodes, oc.WatchPods, oc.WatchServices, oc.WatchEndpoints,
-		oc.WatchNamespaces, oc.WatchNetworkPolicy} {
+// Run starts the actual watching. Also initializes any local structures needed.
+func (oc *Controller) Run(stopChan chan struct{}) error {
+	_, _, err := util.RunOVNNbctl("--columns=_uuid", "list",
+		"port_group")
+	if err == nil {
+		oc.portGroupSupport = true
+	}
+
+	for _, f := range []func() error{oc.WatchPods, oc.WatchServices, oc.WatchEndpoints, oc.WatchNamespaces,
+		oc.WatchNetworkPolicy, oc.WatchNodes} {
```

Comment:
@putnopvut @dcbw  the oc.WatchNodes should be a the beginning. Otherwise, it will reintroduce the bug fixed by this PR https://github.com/ovn-org/ovn-kubernetes/pull/859

Reply:
@girishmg we should probably just split WatchNodes out separately and add a comment to make it clearer.

Reply:
@dcbw I agree. This order has been present from the beginning. I got bitten by changing the order when I removed `net-controller` code and then the PR #859 fixed the order.

Reply:
@girishmg @markmc I fixed that up and force-pushed to this PR